Clientmap: Define p_nodes_min/max as v3s32 instead of v3s16

paramat opened the pull request:
'cam_pos_nodes -/+ box_nodes_d' can exceed the range of v3s16
when a player is near the world edge using a large view range
This previously caused world to disappear
Create new function getBlocksInViewRange() called from
updateDrawList() and renderMap()
Correct code style throughout updateDrawList() and renderMap()
///////////////////////////////////////////
See #3659 for details.
In the end the fix was simple and lightweight, stating p_nodes_min/max as v3s32, it does not need to be limited afterwards to s16 values.
p_nodes_min.max have to be stated in component format to enable each component to be converted from s16 to s32.
Tested at world corner, confirmed that previously world disappears when at (30927, Y, 30927) and view range is increased beyond 1838. 30927 + 1838 = 32765.

v At world corner with high view range.

![screenshot_20160217_042252](https://cloud.githubusercontent.com/assets/3686677/13099890/1b01c5d2-d52f-11e5-80bb-f2b6202dd72f.png)

est31 commented:
:+1: looks good.

It would probably be an idea to have the code style cleanup in a separate commit, at first glance its easy to mix code style cleanup with "simple" type substitution. But that's just an idea/suggestion, nothing I make my +1 dependent upon.

paramat commented:
Hm yes i have a habit of cleaning up as i code, i find it a little difficult to divide my work. Thanks for looking.

paramat commented:
WIP. A new function is being added as this code is called from renderMap() also.

paramat commented:
Updated:
Create new function getBlocksInViewRange() called from
updateDrawList() and renderMap()
Correct code style throughout updateDrawList() and renderMap()

est31 commented on src/clientmap.cpp:
```diff
@@ -550,11 +526,10 @@ void ClientMap::renderMap(video::IVideoDriver* driver, s32 pass)
+					if (buf->getVertexCount() == 0)
 						errorstream<<"Block ["<<analyze_block(block)
-								<<"] contains an empty meshbuf"<<std::endl;
+							<<"] contains an empty meshbuf"<<std::endl;
```
`<<` still needs spaces around it

est31 commented:
:+1:

paramat commented:
Updated, added spaces around '<<' as requested.
Also added spaces around '+' in lines like this:
```
ScopeProfiler sp(g_profiler, prefix + "drawing blocks", SPT_AVG);
```
Hope this is correct.

proller commented:
why mix 2 lines of change and 100 lines of space changes?
can you split in 2 commits ?

paramat commented:
Too late.

paramat commented:
Will merge soon.

paramat commented:
997be666e37e9494bb92a50521b4b749156a609c
